tests/smoke/conjugate_gradient.cpp

```diff
@@ -298,6 +338,14 @@ void grbProgram( const struct input &data_in, struct output &out ) {
 		return;
 	}
 
+	const double residual_norm = sqrt( out.residual );
+	if( residual_norm >= tolerance ) {
+		// notify caller that the residual is too high
+		std::cerr << "Residual is too high: " << residual_norm << " >= " << tolerance << "\n";
+		out.error_code = 40;
+	}
```

Comment:
In this case, the error code returned should also be `FAILED`

Reply:
modification reverted